lib/parse_date/int_from_string.rb

```diff
@@ -279,7 +293,7 @@ def between_bc_latest_year(date_str)
       "-#{Regexp.last_match(:last)}".to_i if date_str.match(BETWEEN_Yn_AND_Yn_BC_REGEX)
     end
 
-    EARLY_NUMERIC_REGEX = Regexp.new('^\-?\d{1,3}([^\du\-\[]|$)', REGEX_OPTS)
+    EARLY_NUMERIC_REGEX = Regexp.new('^\-?\d{1,3}([^\du\[]|$)', REGEX_OPTS)
```

Comment:
removing the hyphen from the follower expression allows '502-504' to be parsed correctly.